packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,632 @@
+  /// opens and at the same time the [Widget] returned by [closedBuilder] is
+  /// faded out while the container grows to fill the surrounding [Navigator].
+  ///
+  /// The `action` callback provided to the builder can be called to close the
+  /// container.
+  final OpenContainerChildBuilder openBuilder;
+
+  /// Whether the entire closed container can be tapped to open it.
+  ///
+  /// Defaults to true.
+  ///
+  /// When this is set to false the container can only be opened via the
+  /// `action` callback provided to [closedBuilder].
+  final bool tappable;
+
+  @override
+  _OpenContainerState createState() => _OpenContainerState();
+}
+
+class _OpenContainerState extends State<OpenContainer> {
+  final GlobalKey<_HideableState> _hidableKey = GlobalKey<_HideableState>();
+  final GlobalKey _closedBuilderKey = GlobalKey();
+
+  void openContainer() {
+    Navigator.of(context).push(_OpenContainerRoute(
+      closedColor: widget.closedColor,
+      openColor: widget.openColor,
+      closedElevation: widget.closedElevation,
+      openElevation: widget.openElevation,
+      closedShape: widget.closedShape,
+      openBuilder: widget.openBuilder,
+      closedBuilder: widget.closedBuilder,
+      hideableKey: _hidableKey,
+      closedBuilderKey: _closedBuilderKey,
+    ));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _Hideable(
+      key: _hidableKey,
+      child: GestureDetector(
+        onTap: widget.tappable ? openContainer : null,
+        child: Material(
+          clipBehavior: Clip.antiAlias,
+          color: widget.closedColor,
+          elevation: widget.closedElevation,
+          shape: widget.closedShape,
+          child: Builder(
+            key: _closedBuilderKey,
+            builder: (BuildContext context) {
+              return widget.closedBuilder(context, openContainer);
+            },
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/// Controls the visibility of its child.
+///
+/// The child can be in one of three states:
+///
+///  * It is included in the tree and fully visible. (The `placeholder` is null
+///    and `visible` is true.)
+///  * It is included in the tree, but not visible; its size is maintained.
+///    (The `placeholder` is null and `visible` is false.)
+///  * It is not included in the tree. Instead a [SizedBox] of dimensions
+///    specified by `placeholder` is included in the tree. (The value of
+///    `visible` is ignored).
+class _Hideable extends StatefulWidget {
+  const _Hideable({
+    Key key,
+    this.child,
+  }) : super(key: key);
+
+  final Widget child;
+
+  @override
+  State<_Hideable> createState() => _HideableState();
+}
+
+class _HideableState extends State<_Hideable> {
+  /// When non-null the child is replaced by a [SizedBox] of the set size.
+  Size get placeholder => _placeholder;
+  Size _placeholder;
+  set placeholder(Size value) {
+    if (_placeholder == value) {
```

Comment:
Assert value != null here?

Reply:
null is totally ok here, it means "don't use the placeholder" as per doc comment.